### DISCUSSION TITLE
Step by step to create a GOAL task

### DISCUSSION THREAD

**Discussion by valmirosjunior** [facebookresearch/ParlAI#4246] [Q&A]
I'm new in ParlAI, I wold like to know if it's possible and how to create a new task in parlai, using intents, and entities to create a Goal task. So I could create a chatbot using the generated model, I guess.

**Comment by klshuster** [comment]
Hi there, you can take a look at how to add a new task in our docs: https://parl.ai/docs/tutorial_task.html

**Reply by valmirosjunior** [reply]
Thanks a lot for the help.

But I've been trying to understand better how to create a GOAL task, but I haven't been successful.
In the suggested documentation, from what I've seen it shows only the basic cases.
 Looking at the articles of tasks, that have already been created, I couldn't find where is explain how to create the data structure, to be used in the construction of the dataset also how can define the Teacher.
To be a little clearer, let's take the task: `ccpe` as an example. Because I didn't find in any documentation, how this structure works.
So I wanted to know if you have any do
